src/App.tsx

```diff
@@ -17,18 +60,37 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onChangeFilter={handleFilterChange}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoadingTodos
+                ? <Loader />
+                : (
+                  <TodoList
+                    todos={visibleTodos}
+                    selectedTodo={selectedTodo}
+                    onChangeTodo={handleSelectTodo}
+                  />
+                )}
+
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {(selectedTodo) && (
```

Comment:
or you can leave those braces here if they serve some higher purpose
```suggestion
      {selectedTodo && (
```